### PR TITLE
Add initial documentation for StatefulSetSlice feature

### PR DESCRIPTION
**Pull request by pwschuurman**
Add documentation for StatefulSetSlice feature targeting Kubernetes alpha in 1.26

* KEP-3335 Enhancement: https://github.com/kubernetes/enhancements/pull/3336
* KEP-3335 Implementation: https://github.com/kubernetes/kubernetes/pull/112744

### CODE REVIEW

**Comment by sftim** [comment]
/retitle Add initial documentation for StatefulSetSlice feature

**Review by sftim — COMMENTED** [review]
Optionally, update https://kubernetes.io/docs/tutorials/stateful-application/basic-stateful-set/#ordered-pod-creation to link to the new Ordinals section.

LGTM as I see it. Needs tech review.

**Review by soltysh — APPROVED** [review]
/lgtm
from tech pov

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md [hunk 1]
```diff
@@ -714,6 +715,8 @@ Each feature gate is designed for enabling/disabling a specific feature:
   memory-backed volumes (mainly `emptyDir` volumes).
 - `StatefulSetMinReadySeconds`: Allows `minReadySeconds` to be respected by
   the StatefulSet controller.
+- `StatefulSetStartOrdinal`: Allow StatefulSet to control start replica ordinal
+   numbering.
```
Comment (tengqm) [inline content/en/docs/reference/command-line-tools-reference/feature-gates.md:719]:
```suggestion
   numbering. Check [StatefulSet ordinals](/[docs/concepts/workloads/controllers/statefulset/#ordinals) for details.
```

File: content/en/docs/tutorials/stateful-application/basic-stateful-set.md [hunk 2]
```diff
@@ -131,6 +131,11 @@ Notice that the `web-1` Pod is not launched until the `web-0` Pod is
 _Running_ (see [Pod Phase](/docs/concepts/workloads/pods/pod-lifecycle/#pod-phase))
 and _Ready_ (see `type` in [Pod Conditions](/docs/concepts/workloads/pods/pod-lifecycle/#pod-conditions)).
 
+#### Custom replica numbering
+
+Note: To customize the replica numbering of pods in a StatefulSet, see
+[Ordinals](docs/tutorials/stateful-application/basic-stateful-set/#ordinals).
```
Comment (tengqm) [inline content/en/docs/tutorials/stateful-application/basic-stateful-set.md:137]: We usually avoid H4 in the document. Maybe render this as a `note` shortcode would be better?

```suggestion
{{< note >}}
To customize the replica numbering of pods in a StatefulSet, see
[Ordinals](/docs/concepts/workloads/controllers/statefulset/#ordinals).
{{< /note >}}
```

**Comment by krol3** [comment]
Hi @pwschuurman! This PR needs a doc review by **Mon Nov 28th** to get this into the release. Please reach out to required SIGs to get their review. Thank you!

**Comment by pwschuurman** [comment]
@soltysh @sftim @tengqm Thanks for the reviews so far. Can you take another quick look before the docs freeze? Thanks!

**Comment by pwschuurman** [comment]
@sftim Thanks. Updated the formatting of the new section

**Comment by pwschuurman** [comment]
@sftim @soltysh Can you take another quick look before the docs freeze? Thanks!

**Review by soltysh — APPROVED** [review]
/lgtm
from tech pov

**Pull request merged**